include/libcyphal/transport/udp/delegate.hpp

```diff
@@ -220,7 +298,70 @@ class TransportDelegate
 
     CETL_NODISCARD UdpardRxRPCDispatcher& getUdpardRpcDispatcher() noexcept
     {
-        return rpc_dispatcher_;
+        return rx_rpc_dispatcher_;
+    }
+
+    void listenForRxRpcPort(UdpardRxRPCPort& rpc_port, const RequestRxParams& params)
```

Comment:
you could avoid the code duplication here by specializing on a private template method with a "is_request" parameter.